### PR TITLE
[FLINK-29803][table] Populate Scala API source jars 

### PR DESCRIPTION
**Pull request by zentol**
Fixes an issue where the source jars of the Table Scala APIs did not contain anything. We have to jump register the source directory with the build-helper-maven-plugin for them to be picked up by the jar plugin.

Additionally this PR clarifies this requirement in other usages, which made it seem that we only use the plugin for eclipse compatibility.

### CODE REVIEW

File: flink-table/flink-table-api-scala-bridge/pom.xml [hunk 1]
```diff
@@ -122,6 +122,42 @@ under the License.
...
+					<!-- Add src/test/scala to build path -->
+					<execution>
+						<id>add-test-source</id>
+						<phase>generate-test-sources</phase>
+						<goals>
+							<goal>add-test-source</goal>
+						</goals>
+						<configuration>
+							<sources>
+								<source>src/test/java</source>
```
Comment (leonardBang) [inline flink-table/flink-table-api-scala-bridge/pom.xml:154]: Do we need this `src/test/java` ? I though we've added them by `maven-jar-plugin:test-jar`.

Comment (zentol) [reply flink-table/flink-table-api-scala-bridge/pom.xml:154]: In this case no. Probably only required if the test sources include both Scala _and_ Java code.

Comment (zentol) [inline flink-table/flink-table-api-scala-bridge/pom.xml:154]:
```suggestion
```

File: flink-table/flink-table-api-scala/pom.xml [hunk 2]
```diff
@@ -129,6 +129,41 @@ under the License.
...
+					<!-- Add src/test/scala to build path -->
+					<execution>
+						<id>add-test-source</id>
+						<phase>generate-test-sources</phase>
+						<goals>
+							<goal>add-test-source</goal>
+						</goals>
+						<configuration>
+							<sources>
+								<source>src/test/java</source>
```
Comment (zentol) [inline flink-table/flink-table-api-scala/pom.xml:160]:
```suggestion
```

**Review by leonardBang — APPROVED** [review]
Thanks @zentol for the update, LGTM

**Pull request merged**